mathics/core/atoms.py

```diff
@@ -165,9 +165,13 @@ def to_python(self, *args, **kwargs):
 
     def round(self, d=None) -> typing.Union["MachineReal", "PrecisionReal"]:
         if d is None:
-            return MachineReal(float(self.value))
-        else:
-            return PrecisionReal(sympy.Float(self.value, d))
+            d = self.value.bit_length()
+            if d <= machine_precision:
+                return MachineReal(float(self.value))
+            else:
+                # machine_precision / log_2(10) + 1
+                d = int(machine_precision / 3.3219 + 1.0)
```

Comment:
We could probably set this as constant in `mathics.core.numbers` and import the value here.